lib/scorecalculator.rb

```diff
@@ -0,0 +1,76 @@
+class ScoreCalculator
+  def initialize
+    @sheet = []
+  end
+
+  def input(*args)
+    errors(*args)
+    @sheet << args
+    bonuspoints
+    puts "You scored #{args.sum} for frame #{currentframe}. Your overall score so far is #{currentscore}"
+    specialannouncement
+    reset if completegame?
+  end
+
+  def scoreboard
+    score = 0
+    @sheet.map.with_index do |x, i|
+      puts "Frame #{i + 1}: #{x.flatten} | #{x.flatten.sum} | #{score += x.flatten.sum}"
+    end
+  end
+
+  private
+
+  def errors(*args)
+    raise 'Score in the frame cannot exceed 10!' if currentframe < 9 && args.sum > 10
+    raise "Max of 2 score inputs allowed if you haven't striked/spared in frame 10" if currentframe == 9 && args.length > 2 && ((args[0] + args[1]) < 10)
+    raise "Max of 2 score inputs allowed if you haven't striked/spared!" if args.length > 2 && currentframe < 9
+    raise 'Only 1 score input allowed if it includes a strike' if args.include?(10) && currentframe < 9 && args.length > 1
```

Comment:
would this throw an error if someone put in [0, 10] though? which, as I understand it, is a valid bowling input, because it's a spare rather than a strike (it's only a strike if you get them all on the first roll). I'm not a bowling expert though!